File: src/main/java/com/exmple/kyudo_app/infrastructure/external/AssignBelongingGroupRepository.java
package com.exmple.kyudo_app.infrastructure.external;

import com.exmple.kyudo_app.domain.model.BelongingGroup;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface AssignBelongingGroupRepository extends JpaRepository<BelongingGroup, Long> {
    Optional<BelongingGroup> findById(Long groupId);
}

File: src/main/kotlin/com/example/kyudo_app/infrastructure/persistence/repository/AssignBelongingGroupRepository.kt
package com.example.kyudo_app.infrastructure.persistence.repository

import com.example.kyudo_app.infrastructure.persistence.entity.BelongingGroupEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface AssignBelongingGroupRepository : JpaRepository<BelongingGroupEntity, String> {
    override fun findById(groupId: String): Optional<BelongingGroupEntity>
}
